server/database/data-creation.js

```diff
@@ -0,0 +1,196 @@
+      colors.push(newColor);
+    }
+  }
+}
+
+
+const randomNumber = (min, max) => {
+  return Math.floor(Math.random() * (max - min) + min);
+ }
+
+const primaryRecordObjs = (counter) => {
+  let dataHolder = [];
+  for (let index = 0; index < records; index ++) {
+    let record = {};
+    record.id = (counter * 1000) + index + 1;
+    record.name = products[randomNumber(0, products.length)];
+    record.category = categories[randomNumber(0, categories.length)];
+    record.manufacturer = faker.company.companyName();
+    if (record.category === 'Men\'s') {
+      clothingMensRecords(record.id, 'men');
+    };
+    dataHolder.push(record);
+    if (record.category === 'Women\'s') {
+      clothingWomensRecords(record.id, 'women');
+    }
+    if (record.category === 'Young Adult\'s') {
+      clothingTeensRecords(record.id, 'teen');
+    }
+    if (record.category === 'Children\'s') {
+      clothingKidsRecords(record.id, 'kid');
+    }
+  }
+  csvWriteControllers.csvWritePrimary
+    .writeRecords(dataHolder)
+    .then(()=> console.log('wrote primary records'))
+}
+
+const clothingMensRecords = (masterRef) => {
+
+  let dataHolder = [];
+
+  let colorSliceStart = randomNumber(0, (colors.length - 15));
+  let colorsSliceEnd = randomNumber(1, 15);
+  let recordColors = colors.slice(colorSliceStart, colorsSliceEnd);
+
+  for (let index = 0; index < sizes.length; index ++) {
+    let record = {};
+    record.masterRef = masterRef;
+    record.size = sizes[index];
+    record.msrp = faker.commerce.price();
+    record.saleStatus = randomNumber(1, 41);
+    record.stock = randomNumber(0, 1000);
+    for (let k = 0; k < colors.length; k++) {
+      let innerRecord = {};
+      innerRecord.color = colors[k];
+      innerRecord.id = indexMensTable;
+      indexMensTable = indexMensTable + 1;
+      Object.assign(innerRecord, record);
+      dataHolder.push(innerRecord);
+    }
+  }
+  csvWriteControllers.csvWriteMens
+  .writeRecords(dataHolder)
+  .then(()=> console.log('wrote men\'s records'))
+}
+
+const clothingWomensRecords = (masterRef) => {
+
+  let dataHolder = [];
+
+  let colorSliceStart = randomNumber(0, (colors.length - 15));
+  let colorsSliceEnd = randomNumber(1, 15);
+  let recordColors = colors.slice(colorSliceStart, colorsSliceEnd);
+
+  for (let index = 0; index < sizes.length; index ++) {
+    let record = {};
+    record.masterRef = masterRef;
+    record.size = sizes[index];
+    record.msrp = faker.commerce.price();
+    record.saleStatus = randomNumber(1, 41);
+    record.stock = randomNumber(0, 1000);
+    for (let k = 0; k < colors.length; k++) {
+      let innerRecord = {};
+      innerRecord.color = colors[k];
+      innerRecord.id = indexWomensTable;
+      indexWomensTable = indexWomensTable + 1;
+      Object.assign(innerRecord, record);
+      dataHolder.push(innerRecord);
+    }
+  }
+  csvWriteControllers.csvWriteWomens
+    .writeRecords(dataHolder)
+    .then(()=> console.log('wrote women\'s records'))
+}
```

Comment:
Generally as Fredrick said we want to remove console.logs from our commits! Other then that everything looks great!

Reply:
ty ty